src/ctap2/commands/get_assertion.rs

```diff
@@ -0,0 +1,681 @@
+            ) -> Result<Self::Output, Retryable<HIDError>> {
+                match status {
+                    Ok(_) | Err(ApduErrorStatus::ConditionsNotSatisfied) => Ok(()),
+                    _ => Err(Retryable::Error(HIDError::DeviceError)),
+                }
+            }
+        }
+
+        let key_handle = self
+            .allow_list
+            .iter()
+            .find_map(|allowed_handle| {
+                let check_command = GetAssertionCheck {
+                    key_handle: allowed_handle.id.as_ref(),
+                    client_data: &self.client_data,
+                    rp: &self.rp,
+                };
+                let res = dev.send_apdu(&check_command);
+                match res {
+                    Ok(_) => Some(allowed_handle.id.clone()),
+                    _ => None,
+                }
+            })
+            .ok_or(HIDError::DeviceNotSupported)?;
+
+        debug!("sending key_handle = {:?}", key_handle);
+
+        let flags = if self.options.user_presence.unwrap_or(false) {
+            U2F_REQUEST_USER_PRESENCE
+        } else {
+            0
+        };
+        let mut auth_data =
+            Vec::with_capacity(2 * PARAMETER_SIZE + 1 /* key_handle_len */ + key_handle.len());
+
+        auth_data.extend_from_slice(
+            self.client_data
+                .hash()
+                .map_err(|e| HIDError::Command(CommandError::Json(e)))?
+                .as_ref(),
+        );
+        auth_data.extend_from_slice(self.rp.hash().as_ref());
+        auth_data.extend_from_slice(&[key_handle.len() as u8]);
+        auth_data.extend_from_slice(key_handle.as_ref());
+
+        let cmd = U2F_AUTHENTICATE;
+        let apdu = U2FAPDUHeader::serialize(cmd, flags, &auth_data)?;
+        Ok(apdu)
+    }
+
+    fn handle_response_ctap1(
+        &self,
+        status: Result<(), ApduErrorStatus>,
+        input: &[u8],
+    ) -> Result<Self::Output, Retryable<HIDError>> {
+        if Err(ApduErrorStatus::ConditionsNotSatisfied) == status {
+            return Err(Retryable::Retry);
+        }
+        if status.is_err() {
+            return Err(Retryable::Error(HIDError::DeviceError));
+        }
+
+        named!(
+            parse_authentication<(u8, u32)>,
+            do_parse!(user_presence: be_u8 >> counter: be_u32 >> (user_presence, counter))
+        );
+
+        let (user_presence, counter, signature) = match parse_authentication(input) {
+            Ok((input, (user_presence, counter))) => {
+                let signature = Vec::from(input);
+                Ok((user_presence, counter, signature))
+            }
+            Err(e) => {
+                error!("error while parsing authentication: {:?}", e);
+                Err(io::Error::new(
+                    io::ErrorKind::Other,
+                    "unable to parse authentication",
+                ))
+                .map_err(|e| HIDError::IO(None, e))
+                .map_err(Retryable::Error)
```

Comment:
These map_err calls are redundant, right? Can we just return `Retryable::Error`?